### PR TITLE
feat!: Align module with latest Snowflake provider changes (removals and deprecations)

### PR DESCRIPTION
**Pull request by dpnoworyta**
This PR implements **BREAKING CHANGES!!!**

This PR completely rewrites the module, to align it with Snowflake provider latest versions (`>0.90`), implements some code optimizations and changes API (variables) in significant way.

List of changes:
- Switched to `snowflake_grant_ownership` resource instead of provider-removed `snowflake_role_ownership_grant`
- Switched to `snowflake_database_role` module to leverage new `database_roles` mechanism
- `default_roles` and `custom_roles` are now combined and managed by single module
- `create_default_roles` variable was renamed to `create_default_database_roles`
- `roles` variable map received following additions:
  - `all_privileges` - optional, bool
  - `on_all` - optional, bool, defaults to false
  - `on_future` - optional, bool, defaults to false
  - `with_grant_option` - optional, bool
  - `granted_to_database_roles` - optional, string
  - `granted_database_roles` - optional, list of strings

- and got following items removed:
  - `comment`
  - `role_ownership_grant`
  - `granted_roles`
  - `granted_to_users`

### CODE REVIEW

**Comment by dgniewek** [comment]
Please take into account changes from this PR: https://github.com/getindata/terraform-snowflake-database-role/pull/6

File: README.md [hunk 1]
```diff
@@ -15,11 +15,11 @@
 Terraform module for Snowflake stage management.
 
 * Creates Snowflake stage
-* Can create custom Snowflake roles with role-to-role, role-to-user assignments
+* Can create custom Snowflake databse-roles with role-to-role assignments
 * Can create a set of default roles to simplify access management:
     * `READONLY` - granted `USAGE` or `READ` privilages
     * `READWRITE` - granted `WRITE` privilages
-    * `ADMIN` - Full access, including schema options like `url` and `credentials`
+    * `ADMIN` - granted `READ`, `WRITE` privilages (role can be additionally granted with `OWNER` attribute when specified)
```
Comment (moss2k13) [inline README.md:22]:
```suggestion
    * `ADMIN` - granted `READ`, `WRITE` privileges (role can be additionally granted with `OWNER` attribute when specified)
```

File: README.md [hunk 2]
```diff
@@ -44,6 +44,35 @@ module "snowflake_stage" {
 - [Simple](examples/simple) - Basic usage of the module
 - [Complete](examples/complete) - Advanced usage of the module
 
+
+## Breaking changes in v2.x of the module
+Due to breaking changes in Snowflake provider and additional code optimizations, **breaking changes** were introduced in `v2.0.0` version of this module.
+
+Lst of code and variable (API) changes:
```
Comment (moss2k13) [inline README.md:51]:
```suggestion
List of code and variable (API) changes:
```

File: README.md [hunk 3]
```diff
@@ -44,6 +44,35 @@ module "snowflake_stage" {
...
+- Switched to `snowflake_grant_ownership` resource instead of provider-removed `snowflake_role_ownership_grant`
+- Switched to `snowflake_database_role` module to leverage new `database_roles` mechanism
+- `default_roles` and `custom_roles` are now combined and managed by single module
+- `create_default_roles` variable was renamed to `create_default_databse_roles`
```
Comment (moss2k13) [inline README.md:55]:
```suggestion
- `create_default_roles` variable was renamed to `create_default_database_roles`
```

File: variables.tf [hunk 4]
```diff
@@ -74,22 +74,30 @@ variable "url" {
   default     = null
 }
 
-variable "create_default_roles" {
-  description = "Whether the default roles should be created"
+variable "create_default_databse_roles" {
```
Comment (moss2k13) [inline variables.tf:77]:
```suggestion
variable "create_default_database_roles" {
```

File: examples/complete/main.tf [hunk 5]
```diff
@@ -1,43 +1,75 @@
...
 
-  name     = "my_stage"
+  name     = "INGEST"
   schema   = snowflake_schema.this.name
   database = snowflake_database.this.name
 
-  comment = "This is my stage"
+  comment = "This is my ingest stage"
+
+  create_default_databse_roles = true
```
Comment (moss2k13) [inline examples/complete/main.tf:39]:
```suggestion
  create_default_database_roles = true
```

File: locals.tf [hunk 6]
```diff
@@ -4,43 +4,32 @@ locals {
...
     lookup(module.stage_label.descriptors, var.descriptor_name, module.stage_label.id), "/${module.stage_label.delimiter}${module.stage_label.delimiter}+/", module.stage_label.delimiter
   ), module.stage_label.delimiter) : null
-  create_default_roles = module.this.enabled && var.create_default_roles
+
+  schema_object_stage_name = "\"${one(snowflake_stage.this[*].database)}\".\"${one(snowflake_stage.this[*].schema)}\".\"${one(snowflake_stage.this[*].name)}\""
 
   is_internal = var.url == null
 
-  default_roles_definition = {
+  default_roles_definition = var.create_default_databse_roles ? {
```
Comment (moss2k13) [inline locals.tf:12]:
```suggestion
  default_roles_definition = var.create_default_database_roles ? {
```

**Review by moss2k13 — APPROVED** [review]

**Pull request merged**